src/main/scala/com/snowflake/snowpark/MergeClause.scala

```diff
@@ -112,8 +112,8 @@ class NotMatchedClauseBuilder private[snowpark] (
       mergeBuilder.joinExpr,
       mergeBuilder.clauses :+ InsertMergeExpression(
         condition.map(_.expr),
-        assignments.keys.toSeq.map(_.expr),
-        assignments.values.toSeq.map(_.expr)),
+        assignments.toSeq.map(_._1.expr),
+        assignments.toSeq.map(_._2.expr)),
```

Comment:
I see.
To make sure the result of `toSeq` is consistent. please save it in a temp variable and reuse, but not call `toSeq` twice. for example.
```scala
val assignmentsSeq = assignments.toSeq
...
        assignmentsSeq.map(_._1.expr),
        assignmentsSeq.map(_._2.expr)),
...
```